### PR TITLE
Picture Formatting Apis: Borders, Change Image and Box Shadow

### PR DESCRIPTION
**Pull request by juliaroldi**
Add content model apis: 

- Change pictures
ChangePicture

- set Image Borders
Borders

- set Image box shadow 
Shadows

### CODE REVIEW

File: packages/roosterjs-content-model/lib/publicTypes/format/formatParts/BorderFormat.ts [hunk 1]
```diff
@@ -21,4 +21,24 @@ export type BorderFormat = {
...
+
+    /**
+     * Width to be applied in all borders
+     */
+    borderWidth?: string;
+
+    /**
+     * Style to be applied in all borders
+     */
+    borderStyle?: string;
```
Comment (JiuqingSong) [inline packages/roosterjs-content-model/lib/publicTypes/format/formatParts/BorderFormat.ts:38]: These 3 will be conflict with borderLeft/Top/Right/Bottom. We just need to keep one set of them.

File: packages/roosterjs-content-model/lib/formatHandlers/common/boxShadowFormatHandler.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { BoxShadowFormat } from '../../publicTypes/format/formatParts/BoxShadowFormat';
+import { FormatHandler } from '../FormatHandler';
+
+/**
+ * @internal
+ */
+export const boxShadowFormatHandler: FormatHandler<BoxShadowFormat> = {
```
Comment (JiuqingSong) [inline packages/roosterjs-content-model/lib/formatHandlers/common/boxShadowFormatHandler.ts:7]: Need test case

File: packages/roosterjs-content-model/lib/formatHandlers/common/boxShadowFormatHandler.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    parse: (format, element) => {
+        if (element.style?.boxShadow) {
+            format.boxShadow = element.style?.boxShadow;
```
Comment (JiuqingSong) [inline packages/roosterjs-content-model/lib/formatHandlers/common/boxShadowFormatHandler.ts:10]: no need to add "?" here

File: packages/roosterjs-content-model/lib/modelApi/image/applyImageBorderFormat.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import { ContentModelImage } from '../../publicTypes/segment/ContentModelImage';
+import { ptToPx } from '../../formatHandlers/utils/pointsToPixels';
+
+/**
+ * @internal
+ */
+export default function applyImageBorderFormat(
+    image: ContentModelImage,
+    border: Border,
+    isPt?: boolean
```
Comment (JiuqingSong) [inline packages/roosterjs-content-model/lib/modelApi/image/applyImageBorderFormat.ts:11]: Can we just parse the unit from the value automatically? We have a util parseValueWithUnit to do this

File: packages/roosterjs-content-model/lib/modelApi/image/applyImageBorderFormat.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    const format = image.format;
+    const { width, style, color } = border;
+    const borderKey = 'borderTop';
+    const extractedBorder = extractBorderValues(format[borderKey]);
+    const borderColor = extractedBorder.color;
+    const borderWidth = extractedBorder.width;
+    const borderStyle = extractedBorder.style;
+    let borderFormat = '';
+
+    if (width) {
```
Comment (JiuqingSong) [inline packages/roosterjs-content-model/lib/modelApi/image/applyImageBorderFormat.ts:22]: Is 0 also a valid value?

File: packages/roosterjs-content-model/lib/modelApi/image/applyImageBorderFormat.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+        borderFormat = `${borderFormat} solid`;
+    }
+
+    if (color) {
+        borderFormat = `${borderFormat} ${color}`;
+    } else if (borderColor) {
+        borderFormat = `${borderFormat} ${borderColor}`;
+    }
+
+    image.format.borderRadius = '5px';
```
Comment (JiuqingSong) [inline packages/roosterjs-content-model/lib/modelApi/image/applyImageBorderFormat.ts:44]: hard code 5px? Is there any way to remove it by user? e.g. if I don't want any border, I should be able to remove this border radius as well

File: packages/roosterjs-content-model/lib/modelApi/image/applyImageBorderFormat.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+ * @internal
+ */
+export default function applyImageBorderFormat(
+    image: ContentModelImage,
+    border: Border,
+    isPt?: boolean
+) {
+    const format = image.format;
+    const { width, style, color } = border;
+    const borderKey = 'borderTop';
```
Comment (JiuqingSong) [inline packages/roosterjs-content-model/lib/modelApi/image/applyImageBorderFormat.ts:15]: Why always top? Should we just retrieve from each side of existing values?

File: packages/roosterjs-content-model/lib/modelApi/image/applyImageBorderFormat.ts (hunk 5 above)
Comment (juliaroldi) [reply packages/roosterjs-content-model/lib/modelApi/image/applyImageBorderFormat.ts:22]: No, the width is a string, it is passed the value + unit. Ex: 15px

File: packages/roosterjs-content-model/lib/modelApi/image/applyImageBorderFormat.ts (hunk 7 above)
Comment (juliaroldi) [reply packages/roosterjs-content-model/lib/modelApi/image/applyImageBorderFormat.ts:15]: We cannot support different styles for each border, then we just need to check one border

File: packages/roosterjs-content-model/lib/publicApi/image/formatImageWithContentModel.ts [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+/**
+ * @internal
+ */
+export default function formatImageWithContentModel(
+    editor: IExperimentalContentModelEditor,
+    apiName: string,
+    callback: (segment: ContentModelImage) => void
+) {
+    formatSegmentWithContentModel(editor, apiName, (_, __, segment) => {
```
[the remark this answers is not in the document]
Comment (juliaroldi) [reply packages/roosterjs-content-model/lib/publicApi/image/formatImageWithContentModel.ts:13]: I'll check if this is required, but it is possible. 
styleMultipleImages

File: packages/roosterjs-content-model/lib/domUtils/borderValues.ts [hunk 9]
```diff
@@ -1,24 +1,9 @@
+import { Border } from '../publicTypes/interface/Border';
+
 /**
  * A combination of CSS border value.
  * See https://developer.mozilla.org/en-US/docs/Web/CSS/border for more information
  */
```
Comment (JiuqingSong) [inline packages/roosterjs-content-model/lib/domUtils/borderValues.ts:6]: Move this comment as well

File: packages/roosterjs-content-model/lib/modelApi/image/applyImageBorderFormat.ts (hunk 7 above)
Comment (JiuqingSong) [reply packages/roosterjs-content-model/lib/modelApi/image/applyImageBorderFormat.ts:16]: hmm ok. In most cases it should be fine. Just a little bit concern about some corner case such as we paste some image that has different borders in each side. We can solve it when that really happens

File: packages/roosterjs-content-model/lib/modelApi/image/applyImageBorderFormat.ts (hunk 5 above)
Comment (JiuqingSong) [reply packages/roosterjs-content-model/lib/modelApi/image/applyImageBorderFormat.ts:23]: ok

**Review by JiuqingSong — APPROVED** [review]

**Pull request merged**